glean-core/src/lib.rs

```diff
@@ -689,6 +695,41 @@ impl Glean {
         self.is_first_run
     }
 
+    /// Set a debug view tag.
+    ///
+    /// This will return `false` in case `value` is not a valid tag.
+    ///
+    /// When the debug view tag is set pings requests receive a `X-Debug-Id` header with the value of the tag
+    /// and are sent to the ["Ping Debug Viewer"](https://mozilla.github.io/glean/book/dev/core/internal/debug-pings.html).
+    ///
+    /// ## Arguments
+    ///
+    /// * `value` - A valid HTTP header value. Must match the regex: "[a-zA-Z0-9-]{1,20}".
+    pub fn set_debug_view_tag(&mut self, value: &str) -> bool {
+        let debug_view_tag = validate_debug_view_tag(value);
+        if debug_view_tag.is_none() {
+            log::error!(
+                "Debug view tag cannot be set. '{}' is not a valid header value.",
+                value
+            );
+            return false;
+        }
+
+        log::info!("Setting debug view tag to {}.", value);
+        self.debug_view_tag = debug_view_tag;
+        true
+    }
+
+    /// Remove the debug view tag.
+    pub fn clear_debug_view_tag(&mut self) {
+        self.debug_view_tag = None;
+    }
+
+    /// Return the value for the debug view tag or `None` if it hasn't been set.
+    pub fn debug_view_tag(&self) -> Option<&String> {
+        self.debug_view_tag.as_ref()
```

Comment:
I find it a bit confusing to have a member with the same name as a method here.  But maybe this is idiomatic Rust?  @badboy?

Reply:
I just followed the convention I saw for other getters in the Glean rust struct, such as [`start_time`](https://github.com/mozilla/glean/blob/main/glean-core/src/lib.rs#L629).

Reply:
Don't see a problem with that. It's idiomatic to provide "getters" that way
(Making individual fields of a struct "pub" makes them both readable and writable if you own the instance)